plugins/edge/plugin.go

```diff
@@ -41,16 +44,19 @@ type Config struct {
 }
 
 type Plugin struct {
-	ctx         context.Context
-	cancel      context.CancelFunc
-	manager     *plugins.Manager
-	logger      *zerolog.Logger
-	config      *Config
-	topazConfig *topaz.Config
-	syncNow     chan api.SyncMode
+	ctx            context.Context
+	cancel         context.CancelFunc
+	manager        *plugins.Manager
+	logger         *zerolog.Logger
+	config         *Config
+	topazConfig    *topaz.Config
+	syncNow        chan api.SyncMode
+	firstRunSignal chan struct{}
```

Comment:
If we change SetServiceStatus to be a func instead of a method, we do not need to add the firstRunSignal channel, make it a singleton, or pass the app pointer.

We can rely on the edge plugin config `p.config.Enabled` to determine if edge sync is enabled. When false, the plugin will not be activated, as the NOOP implementation will be returned instead.

Reply:
I went ahead and removed all the old checks so we are now just checking for `p.config.Enabled` after the sync completes.